test/unit/mock/editable-storage-logic.spec.ts

```diff
@@ -17,130 +17,278 @@ describe('Mock: Editable storage logic', () => {
+      });
+      expect(await mock.getInt256()).to.equal(value);
+      expect(await mock.getBool()).to.equal(true);
+    });
+
+    it('should be able to set bytes32 and an address', async () => {
+      await mock.setVariables({
+        _bytes32: BYTES32_EXAMPLE,
+        _address: ADDRESS_EXAMPLE,
+      });
+      expect(await mock.getBytes32()).to.equal(BYTES32_EXAMPLE);
+      expect(await mock.getAddress()).to.equal(ADDRESS_EXAMPLE);
+    });
+
+    it('should be able to set an address in a packed storage slot and a simple struct', async () => {
+      const struct = {
+        valueA: BigNumber.from(5555),
+        valueB: false,
+      };
+      await mock.setVariables({
+        _packedB: ADDRESS_EXAMPLE,
+        _simpleStruct: struct,
+      });
+
+      expect(await mock.getPackedAddress()).to.equal(ADDRESS_EXAMPLE);
+      expect(convertStructToPojo(await mock.getSimpleStruct())).to.deep.equal(struct);
+    });
+
+    it('should be able to set a uint256 mapping value and a nested uint256 mapping value', async () => {
+      const mapKey = 1234;
+      const mapKeyB = 4321;
+      const nestedMapKey = 1122;
+      const nestedMapKeyB = 3344;
+      const mapValue = 5678;
+      const mapValue2 = 8765;
+      await mock.setVariables({
+        _uint256Map: {
+          [mapKey]: mapValue,
+          [mapKeyB]: mapValue2,
+        },
+        _uint256NestedMap: {
+          [mapKey]: {
+            [nestedMapKey]: mapValue,
+            [nestedMapKeyB]: mapValue2,
+          },
+        },
+      });
+
+      expect(await mock.getUint256MapValue(mapKey)).to.equal(mapValue);
+      expect(await mock.getUint256MapValue(mapKeyB)).to.equal(mapValue2);
+      expect(await mock.getNestedUint256MapValue(mapKey, nestedMapKey)).to.equal(mapValue);
+      expect(await mock.getNestedUint256MapValue(mapKey, nestedMapKeyB)).to.equal(mapValue2);
+    });
+
+    it('should be able to set values in a address => address mapping and a value that was set in the constructor', async () => {
+      const mapKey = ADDRESS_EXAMPLE;
+      const mapValue = '0x063bE0Af9711a170BE4b07028b320C90705fec7C';
+      await mock.setVariables({
+        _addressToAddressMap: {
+          [mapKey]: mapValue,
+        },
+        _constructorUint256: 1234,
+      });
+
+      expect(await mock.getAddressToAddressMapValue(mapKey)).to.equal(mapValue);
+      expect(await mock.getConstructorUint256()).to.equal(1234);
+    });
+
+    it('should be able to set all variables of a contract', async () => {
+      let counterFactory: MockContractFactory<Counter__factory>;
+      let mockCounter: MockContract<Counter>;
+      counterFactory = await smock.mock('Counter');
+      mockCounter = await counterFactory.deploy(1);
+
+      await mockCounter.setVariables({
+        deployer: ADDRESS_EXAMPLE,
+        count: 3,
+      });
+
+      expect(await mockCounter.deployer()).to.equal(ADDRESS_EXAMPLE);
+      expect(await mockCounter.count()).to.equal(3);
+    });
```

Comment:
i'd remove this one and the `Counter` types and factory imports!
the next text is good enough